src/DataFramesMeta.jl

```diff
@@ -376,7 +376,12 @@ end
 function transform(d::AbstractDataFrame; kwargs...)
     result = copy(d)
     for (k, v) in kwargs
-        result[!, k] = isa(v, Function) ? v(d) : v
+        res = isa(v, Function) ? v(d) : v
+        if isa(res, AbstractArray)
+            result[!, k] = res
+        else
+            result[!, k] .= res
```

Comment:
Good point for now. As I have noted in DataFrames.jl these rules will change a bit before 1.0 release, so what we do here is temporary (still `Ref` is probably better as a half measure for now).

Reply:
I made this change and now Tuples work, but can no longer get a `Ref()` test to pass.

Also, I am not really sure it is the right thing to do. `@transform` is more similar to `setindex` than the DataFrame constructor, and setindex requires tuples be wrapped in a `Ref()` to work (as `@transform` did before the change):
```julia
df[!, :C] .= (1,2,3) # errors
df[!, :C] .= Ref((1,2,3)) # works in the same way as the constructor
```

Reply:
A am not sure either if it should be more like `setindex!` (where we essentially disallow implicit broadcasting) or rather `insertcols!` where we allow it.

That is why I have written earlier that DataFramesMeta.jl wil have to be re-thought after DataFrames.jl 1.0 is released. For now I would implement "the most sensible" version that will allow users to get going in the mean time (I do not use DataFramesMeta.jl for my daily work so it is hard to judge for me what is best; I can help you with making sure that what you propose is in line with DataFrames.jl design).

Reply:
Ok, I think I figured out what was messing me up. Checking for equality of Ref types was not doing what I thought it was doing:
```julia
Ref(x) == Ref(x) # false
xref = Ref(x)
xref == xref # true
```
So I assume that equality for Refs is checking whether they exist at the same memory address (normally ===), not whether they can be dereferenced the same number of times to obtain the same value.

That said, the proposed logic has people needing an extra dereference step that they might not anticipate. So, I would propose that Ref types broadcasts without further `Ref()`-ing and other types broadcasts using the `Ref(other)` logic, similar to how it works for `setindex()`. I think this will be the most intuitive to people and, as you say, this is just a stopgap until the v1 rewrite.

Reply:
OK, well I am satisfied with the current state if it pleases the jury.

I inherited a codebase that uses DataFramesMeta somewhat extensively and recently unfroze the package version requirements because they were becoming untenable. This broadcasting behavior was one of the things that changed/broke code when we jumped versions.

That said, I am biased towards your solution of just not using DataFramesMeta for my own code because it removes a dependency and we don't have to keep up with changing behavior in two packages.

Reply:
Before DataFrames.jl release we will unwrap `Ref` and `0`-dimensional arrays in implicit broadcasting, as this is what people intuitively want, see https://github.com/JuliaData/DataFrames.jl/issues/2086.

Reply:
Looks good to me.

Related to DataFramesMeta, I think in the future it might be most sensible for `@transform` to use the selected assignment operator (`=` or `.=`) via `df[!, :col] op value`, which should resolve this question about how it should behave and would automatically mirror any changes to behavior in DataFrames.

Reply:
In DataFrames.jl 1.0 most likely we will have `transform` function defined so we will have to sync it anyway.

Reply:
Lovely!